core/camel-api/src/main/java/org/apache/camel/CamelContextAware.java

```diff
@@ -36,14 +40,27 @@
     CamelContext getCamelContext();
 
     /**
-     * Set the {@link CamelContext} context if the object is an instance of {@link CamelContextAware}.
+     * Injects the {@link CamelContext}
+     *
+     * @param camelContext the Camel context
      */
-    static <T> T trySetCamelContext(T object, CamelContext camelContext) {
-        if (object instanceof CamelContextAware) {
-            ((CamelContextAware) object).setCamelContext(camelContext);
+    void setCamelContext(CamelContext camelContext);
+
+    /**
+     * Get the {@link CamelContext} adapted to the specialized type.
+     * <p/>
+     *
+     * @param  type the type to adapt to
+     * @return      this {@link CamelContext} adapted to the given type
+     *
+     * @see         CamelContext#adapt(Class)
+     */
+    default <T extends CamelContext> T getCamelContext(Class<T> type) {
```

Comment:
Do we really need this, there is already adapt on camel context. Its just that when we keep adding methods here and there, then we loose oversight what is there and what to use.